server/bundler/assets-writer.js

```diff
@@ -26,6 +25,7 @@ Object.assign( AssetsWriter.prototype, {
 		const self = this;
 
 		compiler.hooks.afterEmit.tap( 'AssetsWriter', compilation => {
```

Comment:
i wonder if we should make this an async tap and only allow things to proceed when the write has completed?

Reply:
see https://github.com/webpack/tapable#usage